main.cpp

```diff
@@ -19,15 +19,33 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter): _rewriter(rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+    void run(const MatchFinder::MatchResult &Result) override 
+    {
+        const auto *expression = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        
+        auto range_replace = CharSourceRange::getCharRange(expression->getLParenLoc(),
+                             expression->getSubExprAsWritten()->getBeginLoc());

```

Comment:
Please describe the difference between the use of `getRParenLoc()` and `getSubExprAsWritten()->getBeginLoc()`.

Reply:
getSubExprAsWritten()->getBeginLoc() - location of the start of the name of the variable
getRParenLoc() - location of the closing parenthesis of the type

For the code:
int main() {
    float foo;
    int i = (int)                     foo ;
    return 0;
}

getSubExprAsWritten()->getBeginLoc():

int main() {
    float foo;
    int i = static_cast<int>(foo) ;
    return 0;
}

getRParenLoc():

int main() {
    float foo;
    int i = static_cast<int>(                     foo) ;
    return 0;
}